connector/spanmetricsconnector/connector.go

```diff
@@ -390,8 +390,32 @@ func (p *connectorImp) addExemplar(span ptrace.Span, duration float64, h metrics
 
 type resourceKey [16]byte
 
+func filterResourceAttr(attr pcommon.Map) pcommon.Map {
+	m := pcommon.NewMap()
+	attr.CopyTo(m)
+
+	s := []string{
```

Comment:
Should this slice be expanded to include further attributes?

Reply:
Perhaps OtelScopeName etc?

Reply:
How were these attributes chosen?

What do you think about making this configurable such that, if unset, then the existing behaviour is maintained, or a sensible set of opinionated resource attributes is used if we strongly believe that using all resource attributes _is a bug_; i.e. using all resource attributes to build the key. If this list of resource attributes is explicitly configured, then only those configured attributes will be used as resource keys.

The reason for the suggestion is that I'm not sure if we should be opinionated about which resource attributes we should use as the key, particularly when this is technically a breaking change.

Reply:
I expect this attribute list to be just enough to identify a unique service (span producer). I expect it to be a reduced subset of span dimensions (default and configured). The component stability is alpha, a breaking change shouldn't be much of a surprise :sweat_smile: 

I could make this configurable, but then users need to be aware of the resource attribute semantic conventions. I'll start with the suggested optimizations :+1:

Reply:
@albertteoh I optimized the lookup and made it all configurable :+1: